### PR TITLE
Handle some errors as user exceptions

### PR DESCRIPTION
**Pull request by AdamVyborny**
https://keboola.atlassian.net/browse/COM-1192
https://keboola.atlassian.net/browse/COM-1154
https://keboola.atlassian.net/browse/COM-1112

### CODE REVIEW

File: src/Api/Batch/BatchRequest.php [hunk 1]
```diff
@@ -115,13 +114,24 @@ private function processResponse(Request $request, int $status, array $body): It
...
+            $errorCode = $body['error']['code'] ?? '';
+            switch ($errorCode) {
+                case 'GenericFileOpenError':
+                    throw new UserException('OneDrive API error: The workbook cannot be opened. Make sure ' .
+                        'nobody is editing it.');
+                case 'UnknownError':
+                    throw new UserException('OneDrive API error: The service is unavailable.');
+                case 'MaxRequestDurationExceeded':
+                    throw new UserException('OneDrive API error: Request took too long.');
+                default:
```
Comment (michaljurecko) [inline src/Api/Batch/BatchRequest.php]: Nie vsetky request sa spracovavaju ako `batch`. Mozno by bolo dobre pridat to sem:
https://github.com/keboola/wr-onedrive/blob/debbcfea7dea3a2b92042e144d44ffc31eae75e8/src/Api/Helpers.php#L72-L75

A pouzit tuto metodu aj tu. Co myslis?

Comment (michaljurecko) [reply src/Api/Batch/BatchRequest.php]: Ten typ tam mozno upravit z `RequestException` na inu, vid kde sa pouziva.

Comment (AdamVyborny) [reply src/Api/Batch/BatchRequest.php]: @webrouse přesunuto

**Review by ondrajodas — APPROVED** [review]
jo, takhle to vypadá 👍

**Pull request merged**